src/inmanta/config.py

```diff
@@ -48,6 +48,18 @@ def optionxform(self, name: str) -> str:
         name = _normalize_name(name)
         return super().optionxform(name)
 
+    def _validate_value_types(self, *, section="", option="", value=""):
+        """
+        Override parent class to get clear exceptions
+        """
+        if not isinstance(section, str):
+            raise TypeError(f"section names must be strings, instead received {section} of type {type(section)}")
+        if not isinstance(option, str):
+            raise TypeError(f"option keys must be strings, instead received {option} of type {type(option)}")
+        if not self._allow_no_value or value:
+            if not isinstance(value, str):
+                raise TypeError(f"option values must be strings, instead received {value} of type {type(value)}")
+
```

Comment:
Should we call the parent at the end? To stay compatible with any potential later addition?

Reply:
I wouldn't. We have the behavior we want now. But I could.